sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +40,352 @@ class ConfigurationSetting(Model):
+            ) and key_value.key.startswith(FeatureFlagConfigurationSetting.key_prefix):
+                return FeatureFlagConfigurationSetting._from_generated(  # pylint: disable=protected-access
+                    key_value
+                )
+            if key_value.content_type.startswith(
+                SecretReferenceConfigurationSetting._secret_reference_content_type  # pylint:disable=protected-access
+            ):
+                return SecretReferenceConfigurationSetting._from_generated(  # pylint: disable=protected-access
+                    key_value
+                )
+
         return cls(
             key=key_value.key,
             label=key_value.label,
-            content_type=key_value.content_type,
             value=key_value.value,
+            content_type=key_value.content_type,
             last_modified=key_value.last_modified,
             tags=key_value.tags,
             read_only=key_value.locked,
             etag=key_value.etag,
         )
+
+    def _to_generated(self):
+        # type: (...) -> KeyValue
+        return KeyValue(
+            key=self.key,
+            label=self.label,
+            value=self.value,
+            content_type=self.content_type,
+            last_modified=self.last_modified,
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class FeatureFlagConfigurationSetting(
+    ConfigurationSetting
+):  # pylint: disable=too-many-instance-attributes
+    """A feature flag configuration value.
+    Variables are only populated by the server, and will be ignored when
+    sending a request.
+    :ivar etag: Entity tag (etag) of the object
+    :vartype etag: str
+    :ivar key:
+    :vartype key: str
+    :ivar value: The value of the configuration setting
+    :vartype value: str
+    :ivar enabled:
+    :vartype enabled: bool
+    :param filters:
+    :type filters: list[dict[str, Any]]
+    :param label:
+    :type label: str
+    :param content_type:
+    :type content_type: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=[], **kwargs):  # pylint: disable=dangerous-default-value
```

Comment:
If we add prefix for user, I would suggest not using the name "key" here.

Reply:
Seems like .net uses "featureId"

Reply:
@xiangyan99 what do you suggest?